src/datajudge/constraints/numeric.py

```diff
@@ -158,3 +158,79 @@ def test(self, engine: sa.engine.Engine) -> TestResult:
         )
         result = deviation <= self.max_absolute_deviation
         return TestResult(result, assertion_text)
+
+
+class NumericPercentile(Constraint):
+    def __init__(
+        self,
+        ref: DataReference,
+        k: float,
+        max_absolute_deviation: Optional[float] = None,
+        max_relative_deviation: Optional[float] = None,
+        name: Optional[str] = None,
+        *,
+        ref2: DataReference = None,
+        expected_percentile: float = None,
```

Comment:
I wonder why `mypy` doesn't complain from this incongruency between annotation and default value. Now, me personally, seems odd default value. Looking into the code in `db_access` I see that an `expected_percentile` is always given. 

Any reason not to move this parameter before the `*` and not ake it default to `None`?

And since I'm on this, the previous line about `ref2` also seems like a typing issue which should be `ref2: DataReference | None = None`. If this is happening in other `Constraint`s then I apologize for that oversight in the past.

Reply:
I'm a little confused about this, too. I believe to vaguely recall that we once thought this would mean that mypy accepts the keyword argument not being used by the caller, but would complain about explicitly setting it to `None`. I might be wrong, though.

> Looking into the code in db_access I see that an expected_percentile is always given.

The function `get_percentile` in `db_access` retrieves the percentile of a given data source. If we use this constraint in a `WithinRequirement`, we expect an an `expected_percentile` to be given - since we need something to compare the factual percentile of this single data source to. If we use this constraint in a `BetweenRequirement`, we don't need a fixed reference value `expected_percentile` but instead can just compare the percentiles of both data sources. Therefore this is an optional parameter.

Reply:
Found the culprit!

https://github.com/Quantco/datajudge/blob/main/pyproject.toml#L58

If you don't like this too much, should we move it into a discussion in a separate GitHub issue?